src/clr-angular/forms/select/select.ts

```diff
@@ -4,24 +4,73 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-import {AfterContentInit, Component, ContentChild, ElementRef, HostListener, ViewChild} from "@angular/core";
+import {isPlatformBrowser} from "@angular/common";
+import {
+    AfterContentInit,
+    Component,
+    ContentChild,
+    ElementRef,
+    HostListener,
+    OnDestroy,
+    PLATFORM_ID,
+    Renderer2,
+    ViewChild
+} from "@angular/core";
+import {Subscription} from "rxjs";
 
 import {POPOVER_HOST_ANCHOR} from "../../popover/common/popover-host-anchor.token";
 import {IfOpenService} from "../../utils/conditional/if-open.service";
 import {TAB} from "../../utils/key-codes/key-codes";
+
+import {ClrOption} from "./option";
 import {ClrOptions} from "./options";
+import {OptionSelectionService} from "./providers/option-selection.service";
+import {SelectDomAdapter} from "./utils/select-dom-adapter";
+import {SelectNoopDomAdapter} from "./utils/select-noop-dom-adapter";
+
+// Fixes build error
+// @dynamic (https://github.com/angular/angular/issues/19698#issuecomment-338340211)
+export const selectDomAdapterFactory = (platformId: Object) => {
+    if (isPlatformBrowser(platformId)) {
+        return new SelectDomAdapter();
+    } else {
+        return new SelectNoopDomAdapter();
+    }
+};
 
 @Component({
     selector: "clr-select",
     templateUrl: "select.html",
-    providers: [IfOpenService, {provide: POPOVER_HOST_ANCHOR, useExisting: ElementRef}],
+    providers: [
+        IfOpenService, {provide: POPOVER_HOST_ANCHOR, useExisting: ElementRef}, OptionSelectionService,
+        {provide: SelectDomAdapter, useFactory: selectDomAdapterFactory, deps: [PLATFORM_ID]}
+    ],
     host: {"[class.clr-select]": "true"}
 })
-export class ClrSelect implements AfterContentInit {
+export class ClrSelect<T> implements AfterContentInit, OnDestroy {
     @ViewChild("input") input: ElementRef;
     @ContentChild(ClrOptions) options: ClrOptions;
+    private subscription: Subscription;
 
-    constructor(private ifOpenService: IfOpenService) {}
+    constructor(private ifOpenService: IfOpenService, private optionSelectionService: OptionSelectionService<T>,
+                private renderer: Renderer2, private domAdapter: SelectDomAdapter) {
+        this.initializeSubscriptions();
+    }
+
+    private initializeSubscriptions(): void {
+        this.subscription = this.optionSelectionService.renderSelectionChanged.subscribe((option: ClrOption<T>) => {
+            this.renderSelection(option);
+        });
+    }
+
+    private renderSelection(selectedOption: ClrOption<T>): void {
+        if (this.input && selectedOption) {
+            this.domAdapter.clearChildren(this.input.nativeElement);
+            const clone: HTMLElement = this.domAdapter.cloneNode(selectedOption.elRef.nativeElement);
+            this.renderer.setAttribute(clone, "contenteditable", "false");
```

Comment:
Do we have a test for this one? _The clone shouldn't have `contenteditable` attribute._

Reply:
Actually `contenteditable="false"` would still have the `contenteditable` attribute. It would just be disabled.

Reply:
Then, it would be still worth testing `contenteditable` has `false` value or not? If we ever modify something here and forget to change this attribute value, that test will fail and tell us why.

Reply:
Absolutely. Will add that test.

Reply:
added the test. good catch @Shijir